File: 파이썬/3.스택/3986.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/3986
'''

n = int(input())
cnt = 0
for i in range(n) :
    m = input()
    length = len(m) 
    l = [m[0]] # l : A
    for j in range(1, length) :
        if len(l) == 0:
            l.append(m[j])
        else :
            if l[-1] == m[j] :
                l.pop()
            else :
                l.append(m[j])          
    if len(l) == 0:
        cnt += 1
print(cnt)

File: C++/3.1 스택 - 초급/3986.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/3986
*/

#include <iostream>
#include <stack>

using namespace std;
 
int n, cnt;
string str;

int main() {

    cin >> n;
    
    for(int i=0; i<n; i++){
        stack<char> s;
        cin >> str;
        int len = str.length();
        s.push(str[0]);
        for(int j=1; j<len; j++){
            if(s.empty()){
                s.push(str[j]);
            }else{
                if(s.top() == str[j]){
                    s.pop();
                }else{
                    s.push(str[j]);
                }
            }
        }
        
        if(s.empty()){
            cnt++;
        }
    }
    
    cout << cnt;
    
    return 0;
}